2_ThreeSum.cpp

```diff
@@ -62,31 +62,31 @@ using namespace std;
 vector<vector<int> > threeSum(vector<int> &num)
 {
 	//sort the array
-	sort(num.begin(), num.end());
+    sort(num.begin(), num.end());
     //ret contains the resultant triplets
     vector<vector<int> > res;
     //tuple format of the result
     vector<int> tuples(3);
     for (int i = 0; i < num.size(); ++i) 

```

Comment:
Small optimization: Do we need to go all the way to num.size()?

Reply:
Thanks for pointing this out. It should be num.size() - 2.